devel/validate-platforms

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+#
+# Validate different platform builds of the final Nextstrain image.
+#
+set -euo pipefail
+
+# Set default values.
+registry=localhost:5000
+tag=latest
+
+# Read command-line arguments.
+while getopts "r:t:" opt; do
+    case "$opt" in
+        r) registry="$OPTARG";;
+        t) tag="$OPTARG";;
+        *) echo "Usage: $0 [-r <registry>] [-t <tag>]" 1>&2; exit 1;;
+    esac
+done
+
+IMAGE="$registry/nextstrain/base:$tag"
+PLATFORMS=(linux/amd64 linux/arm64)
+
+main() {
+    # Check software where downloaded versions are not deterministic at
+    # different points in time (e.g. if "latest" is downloaded).
+
+    local report_dir
+    report_dir="$(mktemp -dt "$(basename "$0")")"
+
+    for platform in "${PLATFORMS[@]}"; do
+        echo "[$platform] Pulling image..."
+        docker pull -q --platform "$platform" "$IMAGE"
+
+        echo "[$platform] Checking that the platform is expected..."
+        check-platform "$platform"
+
+        # Initialize a directory for the report file, ensuring slashes in the
+        # platform name are subdirs.
+        mkdir -p "$(dirname "$report_dir/$platform")"
+
+        # Create a report file for the platform.
+        echo "[$platform] Determining software versions..."
+        docker-run "$platform" bash -c '
+            PS4="\$ "
+            set -x
+
+            nextclade --version
+            augur --version
+            auspice --version
+            pip show evofr | grep Version
```

Comment:
_non-blocking_

As an alternative to `pip show | grep`, we could more directly inspect the version:

```
python3 -c 'from importlib.metadata import version; print(version("evofr"))'
```

Quicker too.

Reply:
I like this! It also avoids the weirdly formatted output due to piping, which I didn't bother fixing to make it look right:

```
$ pip show evofr
$ grep Version
Version: 0.1.16
```

Reply:
Updated in [force-push](https://github.com/nextstrain/docker-base/compare/468cb918971f161fce0f9df54c34770614a2f9ee..c70aa6c2d37aed5572ae20f1fdfbd0c287559bb7).